lib/stencil-init.js

```diff
@@ -1,75 +1,67 @@
 
 const serverConfigModule = require('../server/config');
-const jsonLintModule = require('./json-lint');
+const StencilConfigManager = require('./StencilConfigManager');
 const { DEFAULT_CUSTOM_LAYOUTS_CONFIG } = require('../constants');
 
 class StencilInit {
     /**
      * @param inquirer
-     * @param jsonLint
-     * @param fs
+     * @param stencilConfigManager
      * @param serverConfig
      * @param logger
      */
     constructor({
         inquirer = inquirerModule,
-        jsonLint = jsonLintModule,
-        fs = fsModule,
+        stencilConfigManager = new StencilConfigManager(),
         serverConfig = serverConfigModule,
         logger = console,
     } = {}) {
-        this.inquirer = inquirer;
-        this.jsonLint = jsonLint;
-        this.logger = logger;
-        this.fs = fs;
-        this.serverConfig = serverConfig;
+        this._inquirer = inquirer;
+        this._stencilConfigManager = stencilConfigManager;
+        this._serverConfig = serverConfig;
+        this._logger = logger;
     }
 
     /**
-     * @param {string} dotStencilFilePath
      * @param {object} cliOptions
      * @param {string} cliOptions.normalStoreUrl
      * @param {string} cliOptions.accessToken
      * @param {number} cliOptions.port
      * @returns {Promise<void>}
      */
-    async run(dotStencilFilePath, cliOptions = {}) {
-        const oldStencilConfig = this.readStencilConfig(dotStencilFilePath);
+    async run(cliOptions = {}) {
+        const oldStencilConfig = await this.readStencilConfig();
         const defaultAnswers = this.getDefaultAnswers(oldStencilConfig);
         const questions = this.getQuestions(defaultAnswers, cliOptions);
         const answers = await this.askQuestions(questions);
         const updatedStencilConfig = this.applyAnswers(oldStencilConfig, answers, cliOptions);
-        this.saveStencilConfig(updatedStencilConfig, dotStencilFilePath);
+        await this._stencilConfigManager.save(updatedStencilConfig);
 
-        this.logger.log(
+        this._logger.log(
             'You are now ready to go! To start developing, run $ ' + 'stencil start'.cyan,
         );
     }
 
     /**
-     * @param {string} dotStencilFilePath
      * @returns {object}
      */
-    readStencilConfig(dotStencilFilePath) {
-        if (this.fs.existsSync(dotStencilFilePath)) {
-            const dotStencilFile = this.fs.readFileSync(dotStencilFilePath, { encoding: 'utf-8' });
-            try {
-                // We use jsonLint.parse instead of JSON.parse because jsonLint throws errors with better explanations what is wrong
-                return this.jsonLint.parse(dotStencilFile, dotStencilFilePath);
-            } catch (err) {
-                this.logger.error(
-                    'Detected a broken .stencil file:\n',
-                    err,
-                    '\nThe file will be rewritten with your answers',
-                );
-            }
+    async readStencilConfig() {
+        let parsedConfig;
```

Comment:
`let parsedConfig = {};`
and then just 
`return parsedConfig`

Reply:
there are cases when this._stencilConfigManager.read() returns null (config doesn’t exist) and we should return an empty object in this method instead of null